urlutil.py

```diff
@@ -3,6 +3,11 @@
 import os.path
 import urllib.request
 from urllib.parse import urlparse
+try:
+    import boto3
+    from botocore.handlers import disable_signing
+except ImportError:
+    logging.warn("commit hash param will not work without boto3")
```

Comment:
No one cares about this unless they are doing a firmware update that needs boto3, so maybe don't output this all the time and just complain about it later when you need it.

Reply:
agree'd for sure. Moved to when you attempt to use the parameter in the newest commit.